lib/ts/recipe/multifactorauth/api/implementation.ts

```diff
@@ -96,18 +96,26 @@ export default function getAPIInterface(): APIInterface {
                 });
             }
 
+            // next array is filtered to only include factors that are allowed to be setup or are already setup
+            // we do this because the factor chooser in the frontend will be based on the next array only
+            // we do not simply filter out the factors that are not allowed to be setup because in the case
+            // where user has already setup a factor and not completed it, none of the factors will be allowed to
+            // be setup, and that that will result in an empty next array. However, we want to show the factor
+            // that the user has already setup in that case.
+            const next = nextSetOfUnsatisfiedFactors.factorIds.filter(
+                (factorId) => factorsAllowedToSetup.includes(factorId) || factorsSetUpForUser.includes(factorId)
+            );
+
+            if (next.length === 0 && nextSetOfUnsatisfiedFactors.factorIds.length !== 0) {
+                throw new Error(
+                    `The user is required to complete secondary factors they are not allowed to, because of configuration issues`
```

Comment:
You might want to also print out the vlaues in the nextSetOfUnsatisfiedFactors array.+